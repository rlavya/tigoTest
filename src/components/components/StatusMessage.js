import React, { Component } from "react";
import { red400, lightGreen500, orange500 } from "material-ui/styles/colors";
import CONSTANTS from "./../../config/appConstants";

class StatusMessage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isHideNotification: false,
      layoutConfig: {
        type: "error",
        backgroundColor: red400,
        imageIcon: "/images/icn-failure.png"
      }
    };
  }

  componentDidMount() {
    this.configLayoutData();

    setTimeout(() => {
      if(this.props.onClose) {
        this.props.onClose();
      } else {
        this.setState({ isHideNotification: true });
      }
      
    }, CONSTANTS.TIME_THRESHOLD);
  }

  configLayoutData() {
    let configObj = {};
    switch (this.props.type) {
      case "error":
        configObj.backgroundColor = red400;
        configObj.imageIcon = "/images/icn-failure.png";
        configObj.type = this.props.type;
        break;
      case "success":
        configObj.backgroundColor = lightGreen500;
        configObj.imageIcon = "/images/icn-success-white.png";
        configObj.type = this.props.type;
        break;
      case "warning":
        configObj.backgroundColor = orange500;
        configObj.imageIcon = "/images/icn-warning.png";
        configObj.type = this.props.type;
        break;
      default:
    }
  }

  // Rendering Notification layout
  renderMessageLayout() {
    return (
      <div
        className="max-width-4 flex justify-between py2 px1 light text-white"
        style={{ backgroundColor: this.state.layoutConfig.backgroundColor }}
      >
        <div className="flex text-content">
          <img
            alt=""
            src={this.state.layoutConfig.imageIcon}
            className="mr2 flex-none"
            width="31"
            height="31"
          />
          <div className="h5 regular"> {this.props.text}</div>
        </div>
        <div
          onClick={this.props.onClose}
          dangerouslySetInnerHTML={{ __html: "&times;" }}
        />
      </div>
    );
  }

  render() {
    if (!this.state.isHideNotification) {
      return this.renderMessageLayout();
    } else {
      return null;
    }
  }
}

export default StatusMessage;