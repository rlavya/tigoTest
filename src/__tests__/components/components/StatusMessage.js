import StatusMessage from "../../../components/components/StatusMessage";
import renderer from "react-test-renderer";
import React from "react";

describe('StatusMessage component', () => {

  it('should render error', () => {
    const component = renderer.create(
      <StatusMessage type={"error"}/>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render success', () => {
    const component = renderer.create(
      <StatusMessage type={"success"}/>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render warning', () => {
    const component = renderer.create(
      <StatusMessage type={"warning"}/>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

});
