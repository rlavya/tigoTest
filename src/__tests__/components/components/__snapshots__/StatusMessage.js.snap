// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusMessage component should render error 1`] = `
<div
  className="max-width-4 flex justify-between py2 px1 light text-white"
  style={
    Object {
      "backgroundColor": "#ef5350",
    }
  }
>
  <div
    className="flex"
  >
    <img
      className="mr2 flex-none"
      height="31"
      src="/images/icn-failure.png"
      width="31"
    />
    <div
      className="h5 regular"
    >
       
    </div>
  </div>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "&times;",
      }
    }
    onClick={undefined}
  />
</div>
`;

exports[`StatusMessage component should render success 1`] = `
<div
  className="max-width-4 flex justify-between py2 px1 light text-white"
  style={
    Object {
      "backgroundColor": "#8bc34a",
    }
  }
>
  <div
    className="flex"
  >
    <img
      className="mr2 flex-none"
      height="31"
      src="/images/icn-success-white.png"
      width="31"
    />
    <div />
  </div>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "&times;",
      }
    }
    onClick={undefined}
  />
</div>
`;

exports[`StatusMessage component should render warning 1`] = `
<div
  className="max-width-4 flex justify-between py2 px1 light text-white"
  style={
    Object {
      "backgroundColor": "#ff9800",
    }
  }
>
  <div
    className="flex"
  >
    <img
      className="mr2 flex-none"
      height="31"
      src="/images/icn-warning.png"
      width="31"
    />
    <div />
  </div>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "&times;",
      }
    }
    onClick={undefined}
  />
</div>
`;
